core-bundle/src/Image/Studio/FigureBuilder.php

```diff
@@ -226,8 +228,11 @@ public function fromPath(string $path, bool $autoDetectDbafsPaths = true): self
         $path = Path::isAbsolute($path) ? Path::canonicalize($path) : Path::makeAbsolute($path, $this->projectDir);
 
         // Only check for a FilesModel if the resource is inside the upload path
-        if ($autoDetectDbafsPaths && Path::isBasePath(Path::join($this->projectDir, $this->uploadPath), $path)) {
-            $filesModel = $this->getFilesModelAdapter()->findByPath($path);
+        $isDbafsPathPublic = Path::isBasePath(Path::join($this->webDir, $this->uploadPath), $path);
+        $isDbafsPath = $isDbafsPathPublic || Path::isBasePath(Path::join($this->projectDir, $this->uploadPath), $path);
+
+        if ($autoDetectDbafsPaths && $isDbafsPath) {
+            $filesModel = $this->getFilesModelAdapter()->findByPath($isDbafsPathPublic ? Path::makeRelative($path, $this->webDir) : $path);
```

Comment:
How about using a function so that things are only evaluated when needed?

```suggestion
        $getDbafsPath = static function(): ?string {
            if($Path::isBasePath(Path::join($this->webDir, $this->uploadPath), $path)) {
                return Path::makeRelative($path, $this->webDir);
            }
    
            if(Path::isBasePath(Path::join($this->projectDir, $this->uploadPath)) {
                return $path;
            }
               
            return null;
        }

        if ($autoDetectDbafsPaths && null !== ($dbafsPath = $getDbafsPath())) {
            $filesModel = $this->getFilesModelAdapter()->findByPath($dbafsPath);
```